clash-prelude/src/Clash/Explicit/BlockRam.hs

```diff
@@ -1241,99 +1249,264 @@ trueDualPortBlockRamWrapper clkA enA weA addrA datA clkB enB weB addrB datB =
   trueDualPortBlockRam# clkA enA weA addrA datA clkB enB weB addrB datB
 {-# NOINLINE trueDualPortBlockRamWrapper #-}
 
--- | Primitive of 'trueDualPortBlockRam'.
-trueDualPortBlockRam#, trueDualPortBlockRamWrapper ::
-  forall nAddrs domA domB a .
-  ( HasCallStack
-  , KnownNat nAddrs
-  , KnownDomain domA
-  , KnownDomain domB
-  , NFDataX a
-  )
-  => Clock domA
-  -- ^ Clock for port A
-  -> Signal domA Bool
-  -- ^ Enable for port A
-  -> Signal domA Bool
-  -- ^ Write enable for port A
-  -> Signal domA (Index nAddrs)
-  -- ^ Address to read from or write to on port A
-  -> Signal domA a
-  -- ^ Data in for port A; ignored when /write enable/ is @False@
-
-  -> Clock domB
-  -- ^ Clock for port B
-  -> Signal domB Bool
-  -- ^ Enable for port B
-  -> Signal domB Bool
-  -- ^ Write enable for port B
-  -> Signal domB (Index nAddrs)
-  -- ^ Address to read from or write to on port B
-  -> Signal domB a
-  -- ^ Data in for port B; ignored when /write enable/ is @False@
-
-  -> (Signal domA a, Signal domB a)
-  -- ^ Outputs data on /next/ cycle. If write enable is @True@, the data written
-  -- will be echoed. If write enable is @False@, the read data is returned. If
-  -- port enable is @False@, it is /undefined/.
-trueDualPortBlockRam# clkA enA weA addrA datA clkB enB weB addrB datB
-  | snatToNum @Int (clockPeriod @domA) < snatToNum @Int (clockPeriod @domB)
-  = swap (trueDualPortBlockRamModel labelB clkB enB weB addrB datB labelA clkA enA weA addrA datA)
-  | otherwise
-  =       trueDualPortBlockRamModel labelA clkA enA weA addrA datA labelB clkB enB weB addrB datB
- where
-  labelA = "Port A"
-  labelB = "Port B"
 {-# NOINLINE trueDualPortBlockRam# #-}
 {-# ANN trueDualPortBlockRam# hasBlackBox #-}
-
-
--- | Haskell model for the primitive 'trueDualPortBlockRam#'.
+{-# ANN trueDualPortBlockRam# (
+  let
+    bbName = show 'trueDualPortBlockRam#
+    (   _hasCallStack
+     :> knownNatAddrs
+     :> _knownDomainA
+     :> _knownDomainB
+     :> _nfdataX
+
+     :> clockA
+     :> enaA
+     :> wenaA
+     :> addrA
+     :> datA
+
+     :> clockB
+     :> enaB
+     :> wenaB
+     :> addrB
+     :> datB
+
+     :> Nil
+     ) = CV.indicesI @15
+
+    (   symBlockName
+     :> symDoutA
+     :> symDoutB
+     :> Nil
+     ) = CV.indicesI @3
+  in
+    InlineYamlPrimitive [VHDL] [__i|
+    BlackBox:
```

Comment:
Could you change this to
```suggestion
  in InlineYamlPrimitive [VHDL] [__i|
    BlackBox:
```

Reply:
How does this make it any better? I feel like the indentation is still off. `Blackbox` is now neither aligned with `InlineYamlPrimitive`, nor does it make sense w.r.t. `__i`.

I feel:

```
let
  $block
in
  $block
```

gives a clear way of indenting. And so if anything I feel like it should be

```
  in
    InlineYamlPrimitive [VHDL] [__i|
      BlackBox:
```

Reply:
Having said that, I don't want to make this into a big discussion, I really don't think it's important. I'll apply it if that's what you want.

Reply:
To me my suggestion indents on the points where it encloses:
```
a       let
ab        ...
ac        ...
a       in InlineYamlPrimitive [VHDL] [__i|
ad        BlackBox:
ae        |]) #-}
```
`ab`, `ac` , `ad` and `ae` are children of `a`. `a` is the partial expression `let ... in InlineYamlPrimitive [VHDL] [__i| ...`

```
a       let
ab        ...
ac        ...
a       in
ad        InlineYamlPrimitive [VHDL] [__i|
ae        BlackBox:
a       |]) #-}
```

Here, I feel like `a` is the partial expression `let ... in ... |])`, and `InlineYamlPrimitive` and `BlackBox` are siblings rather than a parent-child relationship. You can solve this by another level of indent as well, but I like to have those two characters for free.

However, it is your code and there is no hard rule for this. I also don't want to make a thing of this.

> `Blackbox` is now neither aligned with `InlineYamlPrimitive`

If by "aligned" you mean "at the same level", why would you want that?

> nor does it make sense w.r.t. `__i`.

Mwoah. I think
```
f =
  g 1234 (h (i +
    j 5678))
```
is an acceptable alternative to
```
f = g 1234 (h (i +
               j 5678)
           )
```
when horizontal space becomes scarce (I padded it with 1234 and 5678 to make it a bit larger, but pretend they're long lines).